### ISSUE TITLE
Using CRF with the functional API? 

### ISSUE THREAD

**Issue by joelkuiper** [farizrahman4u/keras-contrib#132]
I'm trying to use the CRF with the Functional (Model) API, but I'm not sure how to proceed. I keep getting `No data provided for "crf_1". Need data for each key in: ['crf_1']`

The general idea is: 

```
    label_td = TimeDistributed(Dense(params["n_labels"], activation="tanh"))(drop)

    crf = CRF(params["n_labels"])
    labels_out = crf(label_td)

    model = Model(inputs=[data_in], outputs=[labels_out])
    model.compile(loss=crf.loss_function,
                  optimizer="adam",
                  metrics=[crf.accuracy])
```

but that doesn't work. Unfortunately I'm unable to rewrite to a sequential model, because the Masking layer complains (I'm feeding it embeddings directly, instead of using an Embedding layer).

**Comment by linxihui** [comment]
@joelkuiper  Does it work if you replace `crf` by `LSTM`?

**Comment by sujanucsc** [comment]
@joelkuiper Were you able to solve this problem? I am having the same.

**Comment by iliaschalkidis** [comment]
This works, just fine...

```
inputs = Input(shape=(None, input_size))
masked_inputs = Masking(mask_value=0.0)(inputs)
drop_inputs = Dropout(dropout_rate)(masked_inputs)
bilstms = Bidirectional(LSTM(units=hidden_unit_size, return_sequences=True,
                             activation="tanh"))(drop_inputs)
drop_bilstms = Dropout(dropout_rate)(bilstms)
crf = CRF(2, sparse_target=True)
outputs_crf = crf(drop_bilstms)

self._model = Model(inputs=inputs, outputs=outputs_crf)
self._model.compile(optimizer=Adam(lr=lr, clipvalue=5.0),
                    loss=crf.loss_function,
                    metrics=crf.accuracy)
```
